si2.api/Controllers/DocumentsController.cs

```diff
@@ -64,10 +65,31 @@ public async Task<IActionResult> UploadDocument(IFormFile file, [FromForm]string
                 user.Id,
                 ct);
 
-            return CreatedAtRoute("GetDataflow", new { id = documentToReturn.Id }, documentToReturn);
+            return CreatedAtRoute("GetDocument", new { id = documentToReturn.Id }, documentToReturn);
         }
 
 
+
+        [HttpGet("{id}", Name = "GetDocument")]
+        [Authorize(AuthenticationSchemes = "Bearer")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(DocumentDto))]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<IActionResult> GetDocument([FromRoute] Guid id, CancellationToken ct)
+        {
+            if (!await _documentService.ExistsAsync(id, ct))
+                return NotFound();
+
+            var document = await _documentService.GetDocumentByIdAsync(id, ct);
+
+            if (await _documentService.IsDeletedAsync(id, ct) == true)
+                return NotFound();
```

Comment:
the check on isDeletedAsync should be in the same if condition as the ExistsAsync before getting the document